test/format/format.sh

```diff
@@ -195,30 +197,85 @@ config=$(find_file "$config")
 	exit 1
 }
 
-verbose "$name configuration: [-c $config] [-h $home]\
-[-j $parallel_jobs] [-n $total_jobs] [-t $minutes] $format_args"
+verbose "$name configuration: $format_binary [-c $config]\
+[-h $home] [-j $parallel_jobs] [-n $total_jobs] [-t $minutes] $format_args"
 
 failure=0
 success=0
+running=0
+status="format.sh-status"
+
+# Report a failure.
+# $1 directory name
+report_failure()
+{
+	dir=$1
+	log="$dir.log"
+
+	echo "$name: failure status reported" > $dir/$status
+	failure=$(($failure + 1))
+
+	# Forcibly quit if first-failure configured.
+	[[ $first_failure -ne 0 ]] && force_quit=1
+
+	echo "$name: job in $dir failed"
+	echo "$name: $dir log:"
+	sed 's/^/    > /' < $log
+}
+
 # Resolve/cleanup completed jobs.
 resolve()
 {
+	running=0
 	list=$(ls $home | grep '^RUNDIR.[0-9]*$')
 	for i in $list; do
 		dir="$home/$i"
 		log="$dir.log"
 
+		# Skip directories that aren't ours.
+		[[ ! -f "$log" ]] && continue
+
 		# Skip failures we've already reported.
-		[[ -f "$dir/reported" ]] && continue
+		[[ -f "$dir/$status" ]] && continue
 
-		# Discard successful jobs.
+		# Get the process ID, ignore any jobs that aren't yet running.
+		pid=`grep -E 'process.*running' $log | awk '{print $3}'`
+		[[ "$pid" =~ ^[1-9][0-9]*$ ]] || continue
+
+		# Leave any process waiting for a gdb attach running, but report it as a failure.
+		grep -E 'waiting for debugger' $log > /dev/null && {
+			report_failure $dir
+			continue
+		}
+
+		# If the job is still running, ignore it unless we're forcibly quitting.
+		kill -s 0 $pid > /dev/null 2>&1 && {
+			[[ $force_quit -eq 0 ]] && {
+				running=$((running + 1))
+				continue
+			}
+			kill -s TERM $pid
```

Comment:
If we're forcibly quitting why not use SIGKILL?

Reply:
I guess I could use `SIGKILL`, but I don't see any advantage to doing so, and I think it makes `format.sh` a bit more complicated (there's another case to handle). Let me know if I'm missing something?